src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -32,6 +32,8 @@ public function __construct(array $decoders = array())
 
     /**
      * {@inheritdoc}
+     * 
+     * @throws RuntimeException If no decoder is found.
```

Comment:
It looks like we talk about different things at the same time.

> How does that break encapsulation?
I was talking about catching subclasses of `SerializerException`: you either break encapsulation (coupling with implementation details) or an interface is bad-designed. If you didn't mean it, I'm sorry.

`@throws SerializerException` means only that serializer failed. Why did it fail, because user input data is invalid or because configuration is wrong? Nobody knows and it doesn't help handle these cases differently. That's what I call "handle differently".